coverage/docs/design.md

```diff
@@ -0,0 +1,113 @@
+#Overview
+This code coverage tool has two major features.
+1. As a pre-submit tool, it runs code coverage on every single commit to Github and reports coverage change back to the PR as a comment by a robot account. It also has the ability to block a PR from merging if coverage falls below threshold
```

Comment:
I still think this needs development. What is being built? Here are my questions:

 - what is a `"pre-submit tool"`? Are you building something that specifically runs pre-submit? Are you just building a code coverage measurement tool? Split the runtime configuration out from _what_ you are building
 - the comments on running for every single commit and reporting and blocking merges -- are you building that? Are you using Prow's built-in job triggering and reporting? Write about what _your_ work will be doing

Reply:
Done.